File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String str = Utils.getAppMetaData(this,"UMENG_CHANNEL");
        Log.d("ddebug","str = " + str);
    }

    public void click(View v){
        startActivity(new Intent(this,MainActivity2.class));
    }
    public void MainActivity3(View v){
        startActivity(new Intent(this,MainActivity3.class));
    }
    public void MainActivity4(View v){
        startActivity(new Intent(this,MainActivity4.class));
    }
    public void MainActivity5(View v){
        startActivity(new Intent(this,MainActivity5.class));
    }
}

File: network/src/main/java/com/example/network/MainActivity.kt
package com.example.network

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.core.app.ActivityCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init();
    }

    private fun init() {
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE), 10001)
        }
    }

    fun demo01(v: View){
        startActivity(Intent(this,Demo01Activity::class.java))
    }

    fun download(v:View){
        startActivity(Intent(this,DownloadActivity::class.java))
    }
}